catalog/to-consul/syncer.go

```diff
@@ -340,36 +377,54 @@ func (s *ConsulSyncer) syncFull(ctx context.Context) {
 
 	// Register all the services. This will overwrite any changes that
 	// may have been made to the registered services.
-	for _, state := range s.nodes {
-		for _, r := range state.Services {
+	for _, services := range s.namespaces {
+		for _, r := range services {
+			if s.EnableNamespaces {
+				// Check and potentially create the service's namespace if
+				// it doesn't already exist
+				err := s.checkAndCreateNamespace(r.Service.Namespace)
+				if err != nil {
+					s.Log.Warn("error checking and creating Consul namespace",
+						"node-name", r.Node,
+						"service-name", r.Service.Service,
+						"consul-namespace-name", r.Service.Namespace,
+						"err", err)
+					continue
+				}
+			}
+
+			// Register the service
 			_, err := s.Client.Catalog().Register(r, nil)
 			if err != nil {
 				s.Log.Warn("error registering service",
 					"node-name", r.Node,
 					"service-name", r.Service.Service,
+					"service", r.Service,
```

Comment:
what does this log line look like. How does it serialize the service object? Wondering if this is usable.

Reply:
For my reference, it looks like:
```
2020-01-29T12:39:17.015-0700 [DEBUG] [Sync] adding service to serviceNames set: service="&{ k8s-sync-632edf7f9889 bar [k8s] map[external-k8s-ns:namespace external-source:k8s] 0  map[] {0 0} false 0 0  <nil> <nil> }" service name=bar
```